hydro_cli/src/core/gcp.rs

```diff
@@ -308,58 +338,83 @@ impl Host for GCPComputeEngineHost {
             }),
         );
 
-        // allow SSH access to all VMs
-        let allow_ssh_rule = format!("{vpc_network}-allow-ssh");
-        firewall_entries.insert(
-            allow_ssh_rule.clone(),
-            json!({
-                "name": allow_ssh_rule,
-                "project": project,
-                "network": format!("${{google_compute_network.{vpc_network}.name}}"),
-                "target_tags": [allow_ssh_rule],
-                "source_ranges": ["0.0.0.0/0"],
-                "allow": [
-                    {
-                        "protocol": "tcp",
-                        "ports": ["22"]
-                    }
-                ]
-            }),
-        );
-
         let vm_instance = format!("vm-instance-{project}-{id}");
-        resource_batch.terraform.resource.entry("google_compute_instance".to_string())
-            .or_default()
-            .insert(vm_instance.clone(), json!({
-                "name": vm_instance,
-                "project": project,
-                "machine_type": self.machine_type,
-                "zone": self.region,
-                "tags": [ allow_ssh_rule ],
-                "metadata": {
-                "ssh-keys": "hydro:${tls_private_key.vm_instance_ssh_key.public_key_openssh}"
-                },
-                "boot_disk": [
-                    {
-                        "initialize_params": [
-                            {
-                                "image": "debian-cloud/debian-11"
-                            }
-                        ]
-                    }
-                ],
-                "network_interface": [
+
+        let mut tags = vec![];
+        let mut external_interfaces = vec![];
+
+        if self.external_ports.is_empty() {
+            external_interfaces.push(json!({
+                "network": format!("${{google_compute_network.{vpc_network}.self_link}}")
+            }));
+        } else {
+            external_interfaces.push(json!({
+                "network": format!("${{google_compute_network.{vpc_network}.self_link}}"),
+                "access_config": [
                     {
-                        "network": format!("${{google_compute_network.{vpc_network}.self_link}}"),
-                        "access_config": [
-                            {
-                                "network_tier": "STANDARD"
-                            }
-                        ]
+                        "network_tier": "STANDARD"
                     }
                 ]
             }));
 
+            // open the external ports that were requested
+            let open_external_ports_rule = format!("{vm_instance}-open-external-ports");
```

Comment:
Could we add an option to open all ports to other services in the same security group?

Reply:
This is already the case! Within a VPC all ports are unblocked.